cmake/SearchForStuff.cmake

```diff
@@ -214,6 +217,12 @@ if (PKG_CONFIG_FOUND)
   if (USE_EXTERNAL_TINYXML)
     message (STATUS "Using system tinyxml.")
     pkg_check_modules(tinyxml tinyxml)
+    if (NOT tinyxml_FOUND)
+      find_package(tinyxml CONFIG)
```

Comment:
Are you sure that this code is actually useful? From the config log, I see:
~~~
2023-08-16T15:47:18.9015616Z -- Checking for module 'tinyxml'
2023-08-16T15:47:18.9368445Z --   Package 'tinyxml', required by 'virtual:world', not found
2023-08-16T15:47:18.9433443Z CMake Warning at vcpkg/scripts/buildsystems/vcpkg.cmake:855 (_find_package):
2023-08-16T15:47:18.9434230Z   Found package configuration file:
2023-08-16T15:47:18.9434627Z 
2023-08-16T15:47:18.9435374Z     D:/a/gazebo-classic/gazebo-classic/vcpkg/installed/x86-windows-release/share/tinyxml/tinyxml-config.cmake
2023-08-16T15:47:18.9436143Z 
2023-08-16T15:47:18.9436475Z   but it set tinyxml_FOUND to FALSE so package "tinyxml" is considered to be
2023-08-16T15:47:18.9437060Z   NOT FOUND.
2023-08-16T15:47:18.9437363Z -- Using system tinyxml2.
2023-08-16T15:47:18.9438008Z Call Stack (most recent call first):
2023-08-16T15:47:18.9438504Z   cmake/SearchForStuff.cmake:221 (find_package)
2023-08-16T15:47:18.9438974Z   CMakeLists.txt:163 (include)
2023-08-16T15:47:18.9439304Z 
2023-08-16T15:47:18.9439420Z 
2023-08-16T15:47:18.9454750Z -- Checking for module 'tinyxml2'
2023-08-16T15:47:19.0120119Z --   Found tinyxml2, version 9.0.0
~~~

The reason for this error is basically the same of tbb, i.e. https://gitlab.kitware.com/cmake/cmake/issues/17135 . You need to unset `tinyxml_FOUND` after `if (NOT tinyxml_FOUND)`. However, as no error message is printed, probably tinyxml is found correctly by the find_path and find_library calls. later on. If everything is already working, perhaps we can drop all the tinyxml related changes?

Reply:
I prefer to do it in the right way. I can unset the found like tbb.

Reply:
There is any advantage in the `find_package(tinyxml)` way? If no, please avoid making changes that could create regression in other (untested) situations, thanks.

Reply:
I revert the changes.